### PR TITLE
mon: only share monmap after authenticating

### PR DESCRIPTION
**Pull request by liewegas**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by smithfarm** [comment]
```
Warning, treated as error:
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/releases/nautilus.rst:document isn't included in any toctree
```

File: doc/releases/nautilus.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+
+
+Upgrading from Luminous
+-----------------------
+
+* During the upgrade from luminous to nautilus, it will not be possible to create
+  a new OSD using a luminous ceph-osd daemon after the monitors have been
+  upgraded to nautilus.
```
Comment (jecluis) [inline doc/releases/nautilus.rst]: I don't find it clear why this happens once this PR is in. Mind explaining the reasoning a bit?

**Comment by liewegas** [comment]
Doh.. put it in PendingReleaseNotes instead

File: doc/releases/nautilus.rst (hunk 1 above)
Comment (liewegas) [reply doc/releases/nautilus.rst]: The luminous MonClient::get_monmap_privately() function sends a MGetMonMap (?) message to the mon to fetch the monmap without authenticating.  That code has been removed from the mon now...

Comment (liewegas) [reply doc/releases/nautilus.rst]: (the only caller for get_monmap_privately() in luminous was ceph-osd during mkfs)

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-08-28_04:25:25-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-29_07:26:03-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-29_11:04:40-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-29_13:22:00-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-29_15:16:19-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/

fixes posted at #23681 and #23803

**Comment by dillaman** [comment]
@liewegas I am seeing some behavior I cannot explain from the rados Python bindings where they fail w/ "wrong fsid" errors after commit 5cf69712f2 (per ```git bisect```). This is when using a cluster w/ CephX disabled. 

Reproducer: 

```
import rados
import json

cluster = rados.Rados(conffile='')
cluster.connect()

cmd = {'prefix': 'osd dump', 'format': 'json'}
rc, buf_s, out = cluster.mon_command(json.dumps(cmd), b'')

print("rc={}, buf={}, out={}".format(rc, buf_s, out))
```

**Comment by dillaman** [comment]
... looks like moving https://github.com/ceph/ceph/commit/5cf69712f239f1c759a3b009ea6ddb7047f9854e#diff-0df37d3bb6424ca661126c4889fb0bffR620 outside the ```if``` clause fixes the issue for me.

**Pull request merged**